### ISSUE TITLE
`nfloat` changes the arguments inside `Max`.

### ISSUE THREAD

**Issue by bgatessucks** [sympy/sympy#10395]
It seems that `nfloat` applied to the product of `Max(symbol, number)` function and another expression changes `number -> 1.0` if `number <= 0`.
Is it me misunderstanding the documentation ?

|expr|result|expected ?|
|:---|:-----|----------|
|`nfloat(Max(y, 0))`|`Max(0, y)`|yes|
|`nfloat(x * Max(y, 0))`|`x*Max(1.0, y)`|no|
|`nfloat(Max(y, -1.1))`|`Max(-1.1, y)`|yes|
|`nfloat(x * Max(y, -1.1))`|`x*Max(1.0, y)`|no|
|`nfloat(Max(y, 1.1))`|`Max(1.1, y)`|yes|
|`nfloat(x * Max(y, 1.1))`|`x*Max(1.1, y)`|yes|

```python
from sympy.functions.elementary.miscellaneous import Max
from sympy import nfloat, symbols

x, y, z = symbols('x y z')
```

**Comment by Arafatk** [comment]
This is clearly a bug 
```
>>> from sympy.functions.elementary.miscellaneous import Max
>>> from sympy import nfloat, symbols
>>> x, y, z = symbols('x y z')
>>> h=nfloat(z* Max(y, -1.1))
>>> print h
z*Max(1.0, y)

>>> h=h/z
>>> print h
Max(1.0, y)

>>> print h.subs(y,0)
1.00000000000000

```

The max of 0,-1.1 is 1 by this function. 
I would like to work on this.

**Comment by Spaethk** [comment]
I need a project for a class, so I am going to attempt to come up with a solution. When I am finished I will submit a pull request. Whichever solution is best may be used in the end.

**Comment by Spaethk** [comment]
I have been working on this issue for a while now and would like to post an update. While doing manual testing to reproduce the bug, I discovered even more unexpected behavior. Basically, if you follow the table posted by the original author... But change 'x' to a symbol with a concrete definition or a number, it causes SymPy to crash!

![table9](https://cloud.githubusercontent.com/assets/10537691/13164974/bfc487da-d67e-11e5-97fe-0e8c52d96d9a.png)

When I remove the following line (2487) from the implementation of 'nfloat()' both the crashes that I had experienced and the issues listed in this bug report cease. However, the tests for this function fail as a result of the change. Can anyone provide any help?

> rv = rv.n(n)

This is the second occurrence of this line in the function. The first occurrence seems to have no negative side effects. I am an amateur when it comes to Python, so any advice is appreciated.
